[PD] Add a Suppressed property to disable a feature

FlachyJoe opened the pull request:
Aims to fix #12073

kadet1090 commented on src/Mod/PartDesign/App/FeatureHole.cpp:
```diff
@@ -1664,6 +1664,10 @@ App::DocumentObjectExecReturn* Hole::execute()
     TopoDS_Shape base;
     try {
         base = getBaseShape();
+        if (Passthrough.getValue()) {
+            this->Shape.setValue(base);
+            return App::DocumentObject::StdReturn;
+        }
```
This gets repeated over and over again - is there a better place for this piece of code so it does not have to be duplicated? That way feature developer need to remember to cover this case which is common for every feature and thus should be higher in the hierarchy.

I am not sure however if this is possible at all now as this probably would have to be done on the caller side?

FlachyJoe replied:
`execute()` is called by `DocumentObject::recompute()` It's not a PD specific procedure so we can't tweak this.

kadet1090 replied:
Yeah, I checked how "Refine" which is also common property is implemented and well, it is also must be called in all features. So this is fine I suppose, maybe an idea for later refactor. I'm marking this as resolved as it is not a PR that should do a major overhaul of how features are implemented.

FlachyJoe replied:
@kadet1090 I found a solution by override `recompute`

FlachyJoe replied:
It doesn't work as expected, I can edit a existing feature but no more create a new one.
Fall back to working previous but still work on this solution.

FEA-eng commented:
It would be awesome to have it. Could it work for FEM features too ? In fact, I was going to create a feature request for that.

FlachyJoe commented:
Rewritten, squashed and rebased. Ready for review.

Side effects:

- if a suppressed feature is used in a pattern, the pattern show its normal result minus the original.
- it's a good manner to reduce a file size.

FlachyJoe commented:
> Could it work for FEM features too ?

The logic is completely different. Feel free to open a FR.

Dutchtallman commented:
> > Could it work for FEM features too ?
> 
> The logic is completely different. Feel free to open a FR.

Why is the logic different? ... to me, the high level concept is the same, but lower level implications are different... similar to 'addition' as a high level concept works out differently on lower level: integers or strings... Should there be generic 'suppress feature'?... with different implementations in different workbenches?
